fireapp/lib/presentation/constraint_form/constraint_form_view.dart

```diff
@@ -0,0 +1,186 @@
+                    }).toList(),
+                    onChanged: (int? newValue) {
+                      setState(() {
+                        viewModel.dropdownValue = newValue!;
+                      });
+                    },
+                  ),
+                ),
+              ],
+            ),
+            const SizedBox(
+              height: 2.0,
+            ),
+            _SchedulerDateInput(
+              viewModel.inputDateController,
+              AppLocalizations.of(context)?.enterDate ?? "",
+              Icons.calendar_today,
+              (v) => v!.isEmpty
+                  ? 'Date is empty!'
+                  : null, //will return this if date is empty when submit
+            ),
+            const SizedBox(
+              height: 2.0,
+            ),
+            _SchedulerTimeInput(
+              viewModel.startTimeController,
+              AppLocalizations.of(context)?.enterStartTime ?? "",
+              Icons.hourglass_top,
+              (v) => v!.isEmpty
+                  ? 'Start Time is empty!'
+                  : null, //will return this if start time is empty when submit
+            ),
+            const SizedBox(
+              height: 420.0,
+            ),
+            ElevatedButton(
+              style: ElevatedButton.styleFrom(
+                minimumSize: const Size.fromHeight(40),
+              ),
+              onPressed: viewModel.submitForm,
+              child: const Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  Text("Add Schedule"),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _SchedulerInputField extends BaseInputField {
+  const _SchedulerInputField(
+    TextEditingController controller,
+    TextStyle? style,
+    String label,
+    IconData icon,
+    String? Function(String?)? validator,
+  ) : super(
+          controller: controller,
+          style: style,
+          label: label,
+          icon: icon,
+          validator: validator,
+          inputType: schedulerInputType.text,
+        );
+}
+
+class _SchedulerDateInput extends BaseInputField {
+  const _SchedulerDateInput(
+    TextEditingController controller,
+    String label,
+    IconData icon,
+    String? Function(String?)? validator,
+  ) : super(
+          controller: controller,
+          label: label,
+          icon: icon,
+          validator: validator,
+          inputType: schedulerInputType.date,
+        );
+}
+
+class _SchedulerTimeInput extends BaseInputField {
+  const _SchedulerTimeInput(
+    TextEditingController controller,
+    String label,
+    IconData icon,
+    String? Function(String?)? validator,
+  ) : super(
+          controller: controller,
+          label: label,
+          icon: icon,
+          validator: validator,
+          inputType: schedulerInputType.time,
+        );
+}
+
+var commonInputBorder = const OutlineInputBorder(
```

Comment:
Avoid these kinds of constants. Just put this in the BaseInputField as a private member for now.